src/java/org/apache/cassandra/net/MessagingService.java

```
@@ -431,6 +431,12 @@ public void shutdown()
 
     public void shutdown(long timeout, TimeUnit units, boolean shutdownGracefully, boolean shutdownExecutors)
     {
+        if (isShuttingDown)
```

I am ok with this, just curious why it was added.  I added logic to ignore the error thrown during jvm-dtest shutdown

```
(IgnoreThrowingRunnable) () -> MessagingService.instance().shutdown(1L, MINUTES, false, true)
```

so only curious what triggered this change?

---

I tried doing similar in https://issues.apache.org/jira/browse/CASSANDRA-15968 but remember hitting race condition issues and eventually dropped doing this; I sadly don't remember what the issues were =(

---

This is because we're doing decommission on the node. You'd hit a "stage is shutdown" exception on "real" node shutdown otherwise.